subsys/net/lib/download_client/src/download_client.c

```diff
@@ -31,9 +31,9 @@ BUILD_ASSERT_MSG(CONFIG_DOWNLOAD_CLIENT_MAX_TLS_FRAGMENT_SIZE <=
 		 CONFIG_DOWNLOAD_CLIENT_MAX_RESPONSE_SIZE,
 		 "The response buffer must accommodate for a full TLS fragment");
 
-#if defined(CONFIG_LOG) && !defined(CONFIG_LOG_IMMEDIATE)
-BUILD_ASSERT_MSG(IS_ENABLED(CONFIG_DOWNLOAD_CLIENT_LOG_HEADERS) ?
-			 CONFIG_LOG_BUFFER_SIZE >= 2048 : 1,
+#if defined(CONFIG_LOG) && !defined(CONFIG_LOG_IMMEDIATE)\
+			&& defined(CONFIG_DOWNLOAD_CLIENT_LOG_HEADERS)
+BUILD_ASSERT_MSG(CONFIG_LOG_BUFFER_SIZE >= 2048,
```

Comment:
Please split this into a separate commit.

Reply:
done